What is a shot worth? Today's notebook will talk about how valuable a shot is, and the most effective way to lead up to those shots.

In [4]:
import pandas as pd


# Use pd.read_csv('FILEPATH')
df = pd.read_csv('C:\\Users\\Elliot\Desktop\\data science datasets\\football-events\\football-events\\events.csv')

df = df.drop(df[df.is_goal != 1].index)


print(len(df[df.shot_place ==3].index))
print(len(df[df.shot_place ==4].index))
print(len(df[df.shot_place ==5].index))
print(len(df[df.shot_place ==12].index))
print(len(df[df.shot_place ==13].index))

7212
6932
4446
2023
2157


7212 shots were made to the bottom left corner, 6932 were made when shot in the bottom right corner, 4446 were aimed the middle, 2023 were made in the top left corner, 2157 were made in top right corner. Now for shots attempted:

In [5]:
df = pd.read_csv('C:\\Users\\Elliot\Desktop\\data science datasets\\football-events\\football-events\\events.csv')

print(len(df[df.shot_place ==3].index))
print(len(df[df.shot_place ==4].index))
print(len(df[df.shot_place ==5].index))
print(len(df[df.shot_place ==12].index))
print(len(df[df.shot_place ==13].index))

18260
18748
25079
5983
6306


So about 1/3 of all shots (on target) in the corners of the goal were made, as opposed to 1/5 in the middle. That is a sizable gap, so it is safe to say that one should shoot in the corners of the goal. Now what about the best way to assist a goal? 

In [7]:
df = pd.read_csv('C:\\Users\\Elliot\Desktop\\data science datasets\\football-events\\football-events\\events.csv')

df = df.drop(df[df.is_goal != 1].index)

print(len(df[df.assist_method ==0].index))
print(len(df[df.assist_method ==1].index))
print(len(df[df.assist_method ==2].index))
print(len(df[df.assist_method ==3].index))
print(len(df[df.assist_method ==4].index))

7877
8662
5143
844
1920


Above are the number of assists by no pass, pass, cross, headed pass, and through ball respectively. So we can see that alot of goals are actually scored by the effort of one person, but most are the result of a pass. Now to see all attempted goals:


In [11]:
df = pd.read_csv('C:\\Users\\Elliot\Desktop\\data science datasets\\football-events\\football-events\\events.csv')
df = df.drop(df[df.event_type != 1].index)    #drop all non-goal attempts


for x in range(0,5):
    print(len(df[df.assist_method == x].index))

61276
109503
43271
7713
7372


According to the results above, a player dribbling by himself and attempting a goal yielded about 13% chance of scoring.

A player that received a pass had a 8% chance of scoring.

A player receiving a cross had a 12% change of scoring.

A player receiving a headed pass had a 11% chance of scoring.

A player receiving a through pass (or through ball) had a whopping 26% chance of scoring.

Takeaways: Given that these were all honest attempts, the through pass had by far the highest chance of resulting in a goal. And since soccer does not penalize offsides heavily, Real Madrid can justify how often Christiano Ronaldo gets caught offsides (given he is the most prolific scorer of our time). And aim for those low corners when shooting!